syncer/main.py

```diff
@@ -54,137 +127,369 @@ def __init__(self, webinterface, connector):
+                                  gateway=self._gateways[ip])
             except Exception as ex:
-                logger.exception('Could not load output mapping: {0}'.format(ex))
+                raise RuntimeError(f'Could not set configuration of remote output {ip}/{remote_id}: {ex}')
+        try:
+            self._call_remote(api_call="set_output", params={
+                "id": remote_id,
+                "is_on": state.get('status'),
+                "dimmer": state.get('dimmer', ""),
+                "timer": state.get('ctimer', "")
+            }, gateway=self._gateways[ip])
+        except Exception as ex:
+            raise RuntimeError(f'Could not set state of remote output {ip}/{remote_id}: {ex}')
 
-        self._headers = {'X-Requested-With': 'OpenMotics plugin: Syncer'}
-        self._endpoint = 'https://{0}/{{0}}'.format(self._ip)
+    def update_remote_shutter_config_and_state(self, ip, local_name, remote_shutter_conf, remote_id, is_shutter_group, local_state, reverse):
+        self.update_remote_shutter_config(ip, remote_shutter_conf=remote_shutter_conf, local_shutter_name=local_name, remote_id=remote_id, is_shutter_group=is_shutter_group)
+        self.update_remote_shutter_state(ip, remote_id, local_state, reverse, is_shutter_group)
 
-        self._enabled = self._ip != '' and self._username != '' and self._password != ''
+    def update_remote_shutter_config(self, ip, remote_shutter_conf, local_shutter_name, remote_id, is_shutter_group):
+        # Set configuration of remote shutter according to local shutter
+        try:
+            if is_shutter_group:
+                shutters_to_rename = []
+                for shutter in remote_shutter_conf:
+                    if shutter.get("group_1") == remote_id or shutter.get("group_2") == remote_id:
+                        shutters_to_rename.append(shutter)
+            else:
+                shutters_to_rename = [remote_shutter_conf[remote_id]]
 
-        logger.info('Syncer is {0}'.format('enabled' if self._enabled else 'disabled'))
+            for shutter in shutters_to_rename:
+                old_name = shutter.get('name')
+                if ' (also controlled by syncer plugin ' in old_name:
+                    old_name = old_name.split(' (also controlled by syncer plugin ')[0]
+                remote_shutter_conf[shutter.get("id")][
+                    'name'] = f"{old_name} (also controlled by syncer plugin {self._name}/{local_shutter_name})"
+                self._call_remote(
+                    f"set_shutter_configuration?config={json.dumps(remote_shutter_conf[shutter.get('id')])}",
+                    gateway=self._gateways[ip])
+        except Exception as ex:
+            raise RuntimeError(
+                f'Could not set configuration remote shutter{"group" if is_shutter_group else ""} {ip}/{remote_id}: {ex}')
+
+    def update_remote_shutter_state(self, ip, remote_id, state, reverse, is_shutter_group):
```

Comment:
is there a good reason why you need to take reverse in account? This should be configured on the gateway where the shutter is connected to

Reply:
The reverse concept you mention is not the same as I use here.
Your concept: 2 outputs in 1 shutter: first 1 is up, second is down; if reverse is enabled, first one is down, second is up
My concept: 2 shutters (shutter 2 follows shutter 1): if shutter 1 goes up, shutter 2 goes up, if shutter 1 goes down, shutter 2 goes down; if reverse is enabled then if shutter 1 goes up, shutter 2 goes down, if shutter 1 goes down, shutter 2 goes up.

Reply:
ok, Don't know why someone would need this, but this is indeed another usecase